cvrf2csaf/cvrf2csaf.py

```diff
@@ -28,30 +42,85 @@ class DocumentHandler:
     CATALOG_FILE = 'schemata/catalog_1_2.xml'
 
     def __init__(self, config):
+
+        self.document_acknowledgments = Acknowledgments()
+        self.document_aggregate_severity = AggregateSeverity()
+        self.comment = Comment()
+        self.document_category = DocumentCategory()
+        self.document_csaf_version = DocumentCsafVersion()
+        self.document_distribution = DocumentDistribution()
+        self.document_lang = DocumentLang()
+        self.document_notes = DocumentNotes()
         self.document_publisher = DocumentPublisher(config['publisher_name'],
                                                     config['publisher_namespace'])
+        self.document_references = DocumentReferences()
+        self.document_source_lang = DocumentSourceLang()
+        self.document_title = DocumentTitle()
         self.document_tracking = DocumentTracking(config['cvrf2csaf_name'],
                                                   config['cvrf2csaf_version'],
                                                   config['force_update_revision_history'])
+        self.product_tree = ProductTree()
+        self.vulnerability = Vulnerability()
+
 
     def _parse(self, root):
         for elem in root.iterchildren():
             # get tag name without it's namespace, don't use elem.tag here
             tag = etree.QName(elem).localname
-            if tag == 'DocumentPublisher':
-                self.document_publisher.create_csaf(elem)
+
+            # ToDo: Lang and SourceLang are missing here.
+
+            if tag == 'Acknowledgments':
```

Comment:
When such dispatch sections look so repetitive and there action statements so redundant if-elif-...-elif-else chains are to me not the obvious implementation.

Why not map with a dict (for the happy paths)and use get to trigger the default (bad) path? I mean, this section looks a lot like a point wise defined mapping i.e. more data than code, right?

Sketching only:
```python
            handler = {
                'Acknowledgements': self.document_acknowledgments,
                'AggregateSeverity': self.document_aggregate_severity,
                'DocumentType': self.document_category,  # rewrite
                'DocumentDistribution': self.document_distribution,
                'DocumentNotes': self.document_notes,
                'DocumentPublisher': self.document_publisher,
                'DocumentReferences': self.document_references,
                'DocumentTitle': self.document_title,
                'DocumentTracking': self.document_tracking,
                'ProductTree': self.product_tree,
                'Vulnerability': self.vulnerability,
            }
            tag_handler = handler.get(etree.QName(elem).localname)
            if tag_handler:
                tag_handler.create_csaf(root_element=elem)
            else:
                logging.warning(f'Not handled input tag {tag}. No parser available.')
```

Reply:
This could be part of a future pull request.